airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/TypeAndDedupeOperationValve.java

```diff
@@ -92,16 +83,12 @@ public void addStream(final AirbyteStreamNameNamespacePair key) {
    * @return a boolean indicating whether we have crossed the interval threshold for typing and
    *         deduping.
    */
-  public boolean readyToTypeAndDedupeWithAdditionalRecord(final AirbyteStreamNameNamespacePair key) {
+  public boolean readyToTypeAndDedupe(final AirbyteStreamNameNamespacePair key) {
      if (!containsKey(key)) {
-     return false;
-     }
-     recordCounts.put(key, recordCounts.get(key) + 1);
```

Comment:
In my testing this only ever gets executed when a batch is ready to flush. On the first flush, it will have not seen the stream so it will add the stream id to the collection and not T&D. On the second batch, as long as it's taken more than 2 minutes it will t&d with presumably the records from the first flush.